sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs

```diff
@@ -981,6 +982,57 @@ public async Task SetAccessControlAsync_ConditionsFail()
             }
         }
 
+        [Test]
+        [ServiceVersion(Min = BlobClientOptions.ServiceVersion.V2019_12_12)]
+        public async Task SetAccessControlRecursiveAsync()
+        {
+            await using DisposingFileSystem test = await GetNewFileSystem();
+            DataLakeDirectoryClient directory = await test.FileSystem.CreateDirectoryAsync(GetNewDirectoryName());
+            DataLakeFileClient file = await directory.CreateFileAsync(GetNewFileName());
+
+            // Act
+            AccessControlChangeResult result = await file.SetAccessControlRecursiveAsync(AccessControlList, null);
+
+            // Assert
+            Assert.AreEqual(0, result.Counters.ChangedDirectoriesCount);
+            Assert.AreEqual(1, result.Counters.ChangedFilesCount);
+            Assert.AreEqual(0, result.Counters.FailedChangesCount);
+        }
+
+        [Test]
+        [ServiceVersion(Min = BlobClientOptions.ServiceVersion.V2019_12_12)]
+        public async Task UpdateAccessControlRecursiveAsync()
+        {
+            await using DisposingFileSystem test = await GetNewFileSystem();
+            DataLakeDirectoryClient directory = await test.FileSystem.CreateDirectoryAsync(GetNewDirectoryName());
+            DataLakeFileClient file = await directory.CreateFileAsync(GetNewFileName());
+
+            // Act
+            AccessControlChangeResult result = await file.UpdateAccessControlRecursiveAsync(AccessControlList, null);
```

Comment:
>UpdateAccessControlRecursiveAsync [](start = 58, length = 33)

It is a bit strange to see "recursive" methods on a file client. I see that these APIs work for file, have we considered renaming the methods for files?

Reply:
File is a legit 1 element tree. The recursive acl operation resemble `chmod -R` command - that one works on files as well as directories without need of special syntax for either one.

Also consider the following scenario. Customer may have in hand a forest of paths - list of PathClients pointing to roots of some trees (for example they might have filtered directory content by some predicate, i.e. all elements starting with some letter). Now they want to loop over that collection and set ACL on every tree in the forest regardless of it's type. If we diverge or remove recursive operation on file then they're going to have quite a lot of work to write such loop instead of having one-liner.
On linux shell it would look like `find . -name a* |  xargs chmod -R a+rw`

Reply:
Makes sense to me. Thanks for the explanation.